stingray/tests/test_base.py

```diff
@@ -47,43 +49,222 @@ def __init__(self, dummy=None):
         StingrayObject.__init__(self)
 
 
-def _check_equal(so, new_so):
-    for attr in ["time", "guefus", "pardulas", "panesapa"]:
-        if not hasattr(so, attr):
-            assert not hasattr(new_so, attr)
-            continue
-        so_attr = at if (at := getattr(so, attr)) is not None else []
-        new_so_attr = at if (at := getattr(new_so, attr)) is not None else []
-
-        assert np.allclose(so_attr, new_so_attr)
-
-    for attr in ["mjdref", "pirichitus", "parafritus"]:
-        if not hasattr(so, attr):
-            assert not hasattr(new_so, attr)
-            continue
-        so_attr = getattr(so, attr)
-        new_so_attr = getattr(new_so, attr)
-        assert so_attr == new_so_attr
-
-
 class TestStingrayObject:
     @classmethod
     def setup_class(cls):
-        cls.arr = [4, 5, 2]
+        cls.arr = np.asarray([4, 5, 2])
         sting_obj = DummyStingrayObj(cls.arr)
         sting_obj.pardulas = [3.0 + 1.0j, 2.0j, 1.0 + 0.0j]
+        sting_obj.sebadas = [[0, 1], [2, 3], [4, 5]]
+        sting_obj._sebadas = [[0, 1], [2, 3], [4, 5]]
         sting_obj.pirichitus = 4
         sting_obj.parafritus = "bonus!"
         sting_obj.panesapa = [[41, 25], [98, 3]]
         cls.sting_obj = sting_obj
 
+    def test_print(self):
+        print(self.sting_obj)
+
     def test_preliminary(self):
         assert np.allclose(self.sting_obj.guefus, self.arr)
 
     def test_instantiate_without_main_array_attr(self):
         with pytest.raises(RuntimeError):
             BadStingrayObj(self.arr)
 
+    def test_equality(self):
+        ts1 = copy.deepcopy(self.sting_obj)
+        ts2 = copy.deepcopy(self.sting_obj)
+        assert ts1 == ts2
+
+    def test_different_array_attributes(self):
+        ts1 = copy.deepcopy(self.sting_obj)
+        ts2 = copy.deepcopy(self.sting_obj)
+        # Add a meta attribute only to ts1. This will fail
+        ts1.blah = 2
+        assert ts1 != ts2
+
+        # Add a non-scalar meta attribute, but with the same name, to ts2.
+        ts2.blah = [2]
+        assert ts1 != ts2
+
+        # Get back to normal
+        del ts1.blah, ts2.blah
+        assert ts1 == ts2
+
+        # Add a non-scalar meta attribute to both, just slightly different
+        ts1.blah = [2]
+        ts2.blah = [3]
+        ts1 != ts2
```

Comment:
should there be an assert statement here?

Reply:
Oops